extensions/iceberg/src/main/java/io/deephaven/iceberg/util/BuildCatalogOptions.java

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.immutables.value.Value;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The options to use with {@link IcebergTools#createAdapter(BuildCatalogOptions)}.
+ */
+@Value.Immutable
+public abstract class BuildCatalogOptions {
+
+    public static Builder builder() {
+        return ImmutableBuildCatalogOptions.builder();
+    }
+
+    /**
+     * The catalog name.
```

Comment:
Borrowing from documentation in IcebergTools::createAdapter
```suggestion
     * The catalog name; if omitted, the catalog URI will be used to generate a name
```